test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -983,6 +983,11 @@ bool UcxConnection::recv_am_data(void *buffer, size_t length, ucp_mem_h memh,
     return process_request("ucp_am_recv_data_nbx", sp, callback);
 }
 
+void UcxConnection::iomsg_recv_defer(std::string iomsg, size_t iomsg_length)
+{
+    _iomsg_recv_backlog.push_back(iomsg.substr(0, iomsg_length));
```

Comment:
IMO we can't use substr to copy buffer since it can be partitially copied if it contains '\0' value in the middle

Reply:
i think std::string does not care about \0